src/Nethermind/Nethermind.Serialization.Rlp/BlockBodyDecoder.cs

```diff
@@ -0,0 +1,293 @@
+// SPDX-FileCopyrightText: 2022 Demerzel Solutions Limited
+// SPDX-License-Identifier: LGPL-3.0-only
+
+using System;
+using System.Buffers;
+using System.Collections.Generic;
+using Nethermind.Core;
+
+namespace Nethermind.Serialization.Rlp
+{
+    public class BlockBodyDecoder : IRlpValueDecoder<BlockBody>, IRlpStreamDecoder<BlockBody>
```

Comment:
What is the difference between these decoders and existing one?

Reply:
I will refactor this.

Reply:
You could use it in BlockDecoder, also we have a `BlockBodyDecoder` in the networking part, maybe they are similar.